getting-started/CONCEPTS/USER-AUTH.md

```diff
@@ -0,0 +1,179 @@
+
+## Configuring the application
+
+Adding user authentication is as simple as configuring it in the `Application` class constructor or builder:
+
+**C#**
+```cs
+AuthenticationOptions<AppState> options = new();
+options.AddAuthentication("graph", new OAuthSettings()
+    {
+        ConnectionName = config.OAUTH_CONNECTION_NAME,
+        Title = "Sign In",
+        Text = "Please sign in to use the bot.",
+    }
+);
+
+Application<AppState> app = new ApplicationBuilder<AppState>()
+    .WithStorage(storage)
+    .WithTurnStateFactory(() => new AppState())
+    .WithAuthentication(adapter, options)
+    .Build();
+```
+
+**Javascript**
+```js
+const app = new ApplicationBuilder<ApplicationTurnState>()
+    .withStorage(storage)
+    .withAuthentication(adapter, {
+        settings: {
+            graph: {
+                connectionName: process.env.OAUTH_CONNECTION_NAME ?? '',
+                title: 'Sign in',
+                text: 'Please sign in to use the bot.',
+            }
+        }
+    })
+    .build();
+```
+
+The `adapter` is the configured `BotAdapter` for the application. The second parameter in the `.withAuthentication` is the authentication options. 
+
+The `settings` property is an object of all the different services that the user could be authenticated to, called *connections*. The above example has the `graph` connection which specifies configurations to authenticate the user to Microsoft Graph. The name `graph` is arbitrary and is used when specifying which service to sign the user in and out of.
+
+The `connectionName` property is what you configure in the Azure Bot Resource, see [Configure OAuth connection for your bot resource](https://learn.microsoft.com/en-us/microsoftteams/platform/bots/how-to/authentication/bot-sso-register-aad?tabs=windows#configure-oauth-connection-for-your-bot-resource).
+
+The `text` property is the titie and the `text` property is the body of the sign in card sent to the user.
+
+### Auto sign in
+
+With this configuration, the bot will attempt to authenticate the user when they try to interact with it. To control when for which incomming activities the bot should authenticate the user, you can specify configure the auto sign in property in the options.
+
+**C#**
+```cs
+options.AutoSignIn = (ITurnContext turnContext, CancellationToken cancellationToken) =>
+ {
+     string command = (turnContext.Activity.Value as JObject).Value<string>("commandId");
+     bool signOutActivity = command == "signOutCommand";
+     if (signOutActivity)
+     {
+         return Task.FromResult(true);
+     }
+     return Task.FromResult(false);
+ };
+```
+
+**Javascript**
+```ts
+.withAuthentication(adapter, {
+    // settings: {
+    //     graph: {
+    //         connectionName: process.env.OAUTH_CONNECTION_NAME ?? '',
+    //         title: 'Sign in',
+    //         text: 'Please sign in to use the bot.',
+    //         endOnInvalidMessage: true
+    //     }
+    // },
```

Comment:
Why is this commented out?

Reply:
the focus in this section is the `autoSignIn` feature

Reply:
The meaning is confusing like this. How about

```
settings: { /* Settings options here... */ }, 
```

That way, the necessity of the options is still apparent but it doesn't clutter up the code example.